In [1]:
import pandas as pd

In [3]:
sales = pd.read_csv("sales.csv")

In [4]:
%%sql
SELECT
    payment,
    AVG(payment_fee) AS avg_payment_fee
FROM
    sales
GROUP BY
    payment;

payment  avg_payment_fee
0  Credit card             0.03
1     Transfer             0.01
2         Cash             0.00

In [5]:
%%sql
SELECT
    CORR(payment_fee, total) AS correlation_order_value,
    CORR(payment_fee, quantity) AS correlation_quantity_ordered
FROM
    sales;

correlation_order_value  correlation_quantity_ordered
0                -0.370509                     -0.429765

In [6]:
%%sql
SELECT
    strftime('%Y-%m', CAST(date AS DATE)) AS month,
    AVG(payment_fee) AS avg_payment_fee
FROM
    sales
GROUP BY
    month
ORDER BY
    month;

month  avg_payment_fee
0  2021-06         0.022574
1  2021-07         0.021942
2  2021-08         0.021514

In [7]:
%%sql
SELECT
    warehouse,
    AVG(payment_fee) AS avg_payment_fee
FROM
    sales
GROUP BY
    warehouse;

warehouse  avg_payment_fee
0     North         0.021676
1   Central         0.022396
2      West         0.021667

In [8]:
%%sql
SELECT
    client_type,
    AVG(payment_fee) AS avg_payment_fee
FROM
    sales
GROUP BY
    client_type;

client_type  avg_payment_fee
0   Wholesale          0.01000
1      Retail          0.02551

In [9]:
%%sql
SELECT
    product_line,
    AVG(payment_fee) AS avg_payment_fee
FROM
    sales
GROUP BY
    product_line;

product_line  avg_payment_fee
0          Miscellaneous         0.020656
1      Electrical system         0.022332
2           Frame & body         0.021988
3  Suspension & traction         0.022368
4                 Engine         0.022787
5        Breaking system         0.021957

In [11]:
%%sql
SELECT
    CASE
        WHEN payment_fee < 0.01 THEN 'Low Fee'
        WHEN payment_fee >= 0.01 AND payment_fee < 0.02 THEN 'Medium Fee'
        ELSE 'High Fee'
    END AS fee_level,
    AVG(order_count) AS avg_order_frequency
FROM (
    SELECT
        client_type,
        COUNT(DISTINCT order_number) AS order_count,
        AVG(payment_fee) AS payment_fee
    FROM
        sales
    GROUP BY
        client_type
) AS subquery
GROUP BY
    fee_level;

fee_level  avg_order_frequency
0   Low Fee                225.0
1  High Fee                775.0

In [12]:
%%sql
SELECT
    CASE
        WHEN payment_fee < 0.01 THEN 'Low Fee'
        WHEN payment_fee >= 0.01 AND payment_fee < 0.02 THEN 'Medium Fee'
        ELSE 'High Fee'
    END AS fee_level,
    AVG(total) AS avg_order_size
FROM
    sales
GROUP BY
    fee_level;

fee_level  avg_order_size
0     Low Fee      165.509483
1    High Fee      167.331669
2  Medium Fee      709.521467

# Payment Fee Analysis

This report presents the findings from the analysis of payment fees in the sales dataset. The analysis aims to understand how payment fees vary across different dimensions and their impact on customer behavior.

## Findings

### 1. Payment Fee Distribution

- The average payment fee varies across different payment methods:
  - Credit card payments: 0.03
  - Transfer payments: 0.01
  - Cash payments: 0.00

### 2. Regional Differences in Payment Fees

- Payment fees differ across warehouse regions:
  - North: 0.021676
  - Central: 0.022396
  - West: 0.021667

### 3. Customer Demographics and Payment Fees

- Wholesale customers have lower average payment fees (0.010) compared to retail customers (0.02551).

### 4. Product Categories and Payment Fees

- Payment fees vary across product categories:
  - Electrical system: 0.022332
  - Breaking system: 0.021957
  - Miscellaneous: 0.020656

### 5. Impact on Customer Behavior

- Contrary to expectations, customers with high payment fees have a higher average order frequency (775.0) compared to those with low payment fees (225.0).
- Customers with medium payment fees have the highest average order size ($709.52), indicating a potential preference for larger purchases despite higher fees.

## Conclusion

The analysis reveals that payment fees exhibit variations across different dimensions such as payment methods, regions, customer demographics, and product categories. Contrary to traditional assumptions, high payment fees do not necessarily deter customers from making purchases, and they may even be associated with higher order frequencies. However, the impact of payment fees on customer behavior is nuanced, with factors such as order size and customer preferences playing significant roles.